src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -157,11 +170,14 @@ private void handleKeyList(RoutingContext rc) {
 
     private void handleRotateMasterKey(RoutingContext rc) {
         try {
-            final RotationResult result = rotateKeys(rc, masterKeyActivatesIn, masterKeyExpiresAfter,
-                    s -> s == Const.Data.MasterKeySiteId || s == Const.Data.RefreshKeySiteId);
+            final RotationResult masterKeyResult = rotateKeys(rc, masterKeyActivatesIn, masterKeyExpiresAfter, masterKeyRotationCutoffTime,
+                s -> s == Const.Data.MasterKeySiteId);
+            final RotationResult refreshKeyResult = rotateKeys(rc, masterKeyActivatesIn, masterKeyExpiresAfter, refreshKeyRotationCutOffTime,
+                    s -> s == Const.Data.RefreshKeySiteId);
```

Comment:
i'm not too familiar with the rotate codes but why do we need to do this second rotateKeys call ofr RefreshSiteId now? or why we didn't need to do this before?

Reply:
Good point - are some keys that never got rotated before starting to get rotated? Didn't think there's such component to this story.

Reply:
The original function does `rotateKeys(rc, masterKeyActivatesIn, masterKeyExpiresAfter, s -> s == Const.Data.MasterKeySiteId || s -> s == Const.Data.RefreshKeySiteId);` which includes both master key and refresh key. 

The reason why I'm doing it twice is because we need to pass in both `masterKeyRotationCutoffTime` and `refreshKeyRotationCutoffTime` to this function and they have different values.

Reply:
ah i see!